### PR TITLE
Concepts Audit Logs, Permissions, Authentification, Colaboration and some re-structuring

### PR DESCRIPTION
**Pull request by hcourdent**

### CODE REVIEW

**Comment by hcourdent** [comment]
Currently fixing Build

File: docs/core_concepts/15_authentification/index.mdx [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    	description="Find out about the roles within a Windmill instance and their respective permissions."
+    	href="/docs/core_concepts/roles_and_permissions"
+    />
+</div>
+
+## Restricted Domain Authentication
+
+Windmill supports authentication [through SSO](../../misc/2_setup_oauth/index.md) for users with email addresses from a restricted domain. This allows organizations to control access to Windmill based on their domain policy. Users with email addresses from the authorized domain can authenticate seamlessly using their SSO credentials.
+
+To enable restricted domain authentication, an administrator can configure the authorized domain in Windmill's settings. Once configured, users with email addresses from the authorized domain will be able to log in using their SSO provider.
```
Comment (rubenfiszel) [inline docs/core_concepts/15_authentification/index.mdx:19]: replace "in Windmill Settings" by, "In the OAuth configuration by setting 'allowed_domains' to the desired domains (e.g: 'windmill.dev' to accept only Google/Microsoft logins with a xxx@windmill.dev address)

File: docs/core_concepts/15_authentification/index.mdx [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+## Manually Add Users to a Windmill Instance
+
+As a superadmin of the workspace, you have the ability to manually add users to the Windmill instance. This is useful for inviting users who do not have SSO credentials or for providing access to individuals outside the restricted domain.
```
Comment (rubenfiszel) [inline docs/core_concepts/15_authentification/index.mdx:23]: As a superadmin of the *instance*

File: docs/core_concepts/15_authentification/index.mdx [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+![Manually Add Users](./add_global_users.png "Manually Add Users")
+
+If SMTP is configured, an email will be sent to the user with their account details and instructions for accessing Windmill.
+
+By default, users are not invited to any workspace, unless auto-invite has been set-up.
+
+## Adding Users to a Workspace
+
+Once added to an instance, users can create their own workspace (on cloud, those ones are not considered as Premium as plans work at the workspace-level). However, by default they will not be invited to any workspace.
```
Comment (rubenfiszel) [inline docs/core_concepts/15_authentification/index.mdx:43]: > (on cloud, those ones are not considered as Premium as plans work at the workspace-level)

remove or clarify, I would remove

Also add that windmill can be configured to remove the ability for non-superadmins to create workspace using the: CREATE_WORKSPACE_REQUIRE_SUPERADMIN env variable.

File: docs/core_concepts/16_roles_and_permissions/index.mdx [hunk 4]
```diff
@@ -0,0 +1,69 @@
+import DocCard from '@site/src/components/DocCard';
+
+# Roles and Permissions
+
+Windmill provides a roles and permissions system that allows you to control access and manage permissions within your instance and workspaces. Different roles have different levels of access to scripts, flows, apps, and other entities in Windmill.
+
+## Roles in Windmill
+
+1. **Superadmin**: The superadmin role has the highest level of access and can manage the entire Windmill instance, including all its workspaces and invites. Superadmins have full control over the configuration and settings of the Windmill instance.
```
Comment (rubenfiszel) [inline docs/core_concepts/16_roles_and_permissions/index.mdx:9]: The superadmin role has the highest level of access and can manage the entire Windmill instance. Only superadmins can access the dedicated 'admins' workspace. Also by default, superadmins have access to all workspaces as admins of those workspaces. When an admin do an action as a user of a workspace he is not a member of and as such as no username in, his email is used in lieu of the username.

File: docs/core_concepts/16_roles_and_permissions/index.mdx [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+2. **Admin**: Admins have the ability to manage a specific Windmill workspace and invites within that workspace. They can see and edit all scripts, flows, apps, and other entities within the workspace. Admins have the power to create, modify, and delete entities and can manage permissions within the workspace.
```
Comment (rubenfiszel) [inline docs/core_concepts/16_roles_and_permissions/index.mdx:11]: and users within that workspace (invites is unecessarily specific)

File: docs/core_concepts/16_roles_and_permissions/index.mdx [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+
+3. **Author**: Authors can execute and view scripts, flows, and apps within a workspace. They also have the privilege of creating new scripts, flows, and apps. Authors have read and write access to the entities they create or have been [granted access to by others](../8_groups_and_folders/index.md).
```
Comment (rubenfiszel) [inline docs/core_concepts/16_roles_and_permissions/index.mdx:13]: Authors see the full interface of windmill and contrary to the Operators, have the ability to create and edit scripts, flows and apps.

File: docs/core_concepts/16_roles_and_permissions/index.mdx [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+
+4. **Operator**: Operators have limited access within a workspace. They can only execute and view scripts, flows, and apps that they have visibility on, and only those that are within their allowed path or [folders](../8_groups_and_folders/index.md). Operators do not have the ability to create or modify entities.
+
+5. **Anonymous app viewers**: Anonymous app viewers are individuals who can access and view Windmill apps by knowing the [secret URL](../../apps/0_toolbar.mdx#deploy) that allows app viewing. They do not have any editing or execution privileges.
```
Comment (rubenfiszel) [inline docs/core_concepts/16_roles_and_permissions/index.mdx:17]: or privileges to execute scripts and flows. The apps that they see execute the scripts and flows that are part of the app but only on behalf of the app author. The anonymous user cannot execute any of the scripts and flows part of the apps outside of the inputs authorized by the app through a policy that is automatically generated by the app.

File: docs/core_concepts/16_roles_and_permissions/index.mdx [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+
+- [groups](../8_groups_and_folders/index.md#groups)
+- [folders](../8_groups_and_folders/index.md#folders)
+- [scripts](../../getting_started/0_scripts_quickstart/index.mdx)
+- [resources](../3_resources_and_types/index.md)
+- [variables](../2_variables_and_secrets/index.md)
+- [schedules](../1_scheduling/index.md)
+- [jobs](../reference#jobs)
+
+Each entity's ACL defines its owner and specifies the read and write permissions. By default, only the owner of an entity has read and write access to it. Entities can be explicitly shared with other groups and users in either read-only or read and write mode through [folders](../8_groups_and_folders/index.md). Write mode implicitly includes read permission. However, it's important to exercise caution when granting write access, as it allows users to delete, rename, and modify the sharing of the entity.
```
Comment (rubenfiszel) [inline docs/core_concepts/16_roles_and_permissions/index.mdx:46]: Write access does not grant rename or modify. You have to be an owner.

So there are actually 3 levels of ACL:
Viewer, Writer, Owner

let's chat about it

File: docs/core_concepts/17_collaboration/index.mdx [hunk 9]
```diff
@@ -0,0 +1,130 @@
+import DocCard from '@site/src/components/DocCard';
+
+# Collaboration in Windmill
+
+Collaboration in Windmill is facilitated through various features and workflows.
```
Comment (rubenfiszel) [inline docs/core_concepts/17_collaboration/index.mdx:5]: facilitated is a faux ami I believe

**Pull request merged**